lib/far_mar/product.rb

```diff
@@ -0,0 +1,44 @@
+class FarMar::Product < FarMar::Base
+  attr_accessor :id, :name, :vendor_id
+  FILE = ("support/products.csv")
+
+  def initialize(attrs)
+   @id= attrs[0].to_i
+   @name = attrs[1]
+   @vendor_id = attrs[2].to_i
+  end
+
+
+  def vendor
+    vendors = FarMar::Vendor.all
+    all_vendors = vendors.group_by do |obj|
+      obj.id
+    end
+    all_vendors[@vendor_id].first
+  end
+
+  def sales
+    sales = FarMar::Sale.all
+    all_sales = sales.group_by do |obj|
+      obj.product_id
+    end
+    all_sales[@id]
+  end
+
+  def number_of_sales
+    all_sales = self.sales
```

Comment:
This method is great, it's cool to see this logical approach to determining the size, but ruby has this idea built in, it could just be `self.sales.count`